### ISSUE TITLE
there are an plan to release the training part?

### ISSUE THREAD

**Issue by XiaoqingNLP** [THUDM/GLM-130B#6]
The stable training contribution mentioned in the document, while its code and script are not released. Is there an open source plan ?

**Comment by Sengxian** [comment]
Thanks for your attention, the training code will be open source after we clean it (it's still messy), probably later this month.

**Comment by XiaoqingNLP** [comment]
@Sengxian 1. Is it convenient to introduce the basic training settings? For example, the number of model parallel `mp` and `zero stage ` in training, the minimum number of nodes required for training?

**Comment by Sengxian** [comment]
We tried numerous combinations of parallelism and ended up using 4-way tensor parallel and 8-way pipeline parallelism for the best training throughput, approximately 135 TFLOPS / GPU. Adam optimizer with ZeRO stage 1 is used in actual training. At least 40 nodes are required to start training. Using optimizers like Adafactor or 8-bit Adam could further reduce the minimum number of nodes required to 24 nodes, however, we did not use them for best convergence.
